src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -66,6 +66,22 @@ private static bool EndsWithPeriodOrSpace(string path)
             return c == ' ' || c == '.';
         }
 
+        /// <summary>
+        /// Adds the extended path prefix (\\?\) if not already a device path, IF the path is not relative,
+        /// AND the path is more than 259 characters. (> MAX_PATH + null)
+        /// </summary>
+        internal static string EnsureExtendedPrefixOverMaxPath(string path)
```

Comment:
What API did this get added for? I specifically updated to do this below to put the prefix on with trailing spaces & periods to fix handling those paths.

Reply:
C:\git\corefx\src\Common\src\Interop\Windows\kernel32\Interop.CreateFile2.cs(95,39): error CS0117: 'PathInternal' does not contain a definition for 'EnsureExtendedPrefixOverMaxPath' [C:\git\corefx\src\System.IO.FileSystem.Watcher\src\System.IO.FileSystem.Watcher.csproj]

Reply:
Thanks- please change that to the method below and remove this method.

Reply:
@JeremyKuhne Ok, so to confirm, I'll replace the call to `EnsureExtendedPrefixOverMaxPath` in Interop.CreateFile2.cs with a call to `EnsureExtendedPrefixIfNeeded` ?